Add warnings w.r.t. third-party toolchain issues on Darwin (#40)
theos/theos.dev: docs/Installation-macOS.md

1. Run the installer:

		bash -c "$(curl -fsSL https://raw.githubusercontent.com/theos/theos/master/bin/install-theos)"

> # Troubleshooting

Reviewer: Given that this isn't exactly installation related, do we want to add a `Troubleshooting.md` page under Help.md where we can add common issues + fixes like this?

Author: Again, we can't assume users will read all the non-critical doc pages, because they don't

Reviewer 2: This page isn't that large and I agree with Naville's comment. Resolving

## Third-party Toolchains

> Some third-party toolchains have issues with [Modules](https://clang.llvm.org/docs/Modules.html) on macOS. In such cases, you may use ``instance_USE_MODULES`` to disable modules.

Reviewer: ```suggestion
Some third-party toolchains have issues with [Modules](https://clang.llvm.org/docs/Modules.html) on macOS. In such cases, you can either try a different third-party toolchain (e.g., a newer or older version) or disable Modules for your project with `XXX_USE_MODULES=0`.
```

> **Example**:  

Reviewer: I don't think an example usage is necessary

Author: I think our community as a whole def needs more detailed example given my past experience

Reviewer 2: I agree, more examples don't hurt.

This doesn't necessarily set a precedent that we need to provide such examples elsewhere. Resolving

```
TWEAK_NAME = Test

> Test_USE_MODULES=no

Reviewer 3: ```suggestion
Test_USE_MODULES = no
```

> Test_USE_MODULES = no

Reviewer: ```suggestion
Test_USE_MODULES = 0
```

We may want to change this so that it's consistent with the syntax given above.
